### ISSUE TITLE
App instantly crashes when opened "FailedMigrationError"

### ISSUE THREAD

**Issue by stoppablemurph** [signalapp/Signal-Android#11350]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Whenever I try to open the Signal app, it immediately crashes

### Steps to reproduce
- Open the Signal app (I don't have steps to reproduce the issue with a working install of Signal, and I don't have any idea what caused it to start happening in the first place.)

**Actual result:** The app immediately crashes back to the Android home screen
**Expected result:** The app should function normally without crashing

### Screenshots
![signal-crash](https://user-images.githubusercontent.com/5564311/120225473-2e6fdd80-c1fa-11eb-9f5a-1acd348cb33d.gif)

### Device info

**Device:** Google Pixel 5
**Android version:** 11 (Build number: RQ2A.210505.003) (The issue was present in the previous OS version as well)
**Signal version:** 5.13.2 (The issue was present in the previous app version as well)

### Link to debug log
I can't run the debug thing from the app for what I hope are obvious reasons. I do have an [adb logcat snippet from when I attempt to open the app](https://github.com/signalapp/Signal-Android/files/6571645/logcat-securesms.log) though.

### Other notes
* It's possible this is a duplicate issue, but looking at other `MissingRecipientException` related issues, and issues in general, I was unable to find anything specifically related to a `FailedMigrationError`
* I've been experiencing this issue for about 10 days as of opening this issue (May 21)
* I use the Signal MacOS and Windows clients as well, which themselves are still currently functional, with only the mobile app not working

### Troubleshooting attempts
I've tried the following thus far with no change in behavior:
* Force stopping and re-opening the app
* Clearing the app cache
* Rebooting the device
* Updating the app

I have NOT tried:
* Restoring from a backup snapshot yet. I expect I will likely have to do so, and will update here with any notable information when I'm done, (though the issue should ideally be addressed regardless of whether or not a backup restore is successful)

**Comment by cody-signal** [comment]
Thanks for all the info, and especially the logcat snippet.

I've put a fix into 5.13.4 that'll not fail this job, but I think you'll run into other issues given the core of the problem being unable to find the model that represents you. Feel free to post here on how it goes.

**Comment by stoppablemurph** [comment]
Sorry I keep forgetting to reply to this. Thanks for the fix! I was able to get things working with a restore to a backup, but I nearly wasn't because at some point the app did something that completely deleted it's database and my on device backups were replaced with essentially empty ones. I was lucky that on a whim I happened to back up the last good back up to another device.

I haven't had any issues since the backup restore though. Since you put out the patch and it doesn't seem to be a widespread issue, I'm going to consider this resolved.
